src/wallet/wallet2.cpp

```diff
@@ -3721,8 +3721,9 @@ void wallet2::light_wallet_get_outs(std::vector<std::vector<tools::wallet2::get_
     THROW_WALLET_EXCEPTION_IF(outs.back().size() < fake_outputs_count + 1 , error::wallet_internal_error, "Not enough fake outputs found" );
     
     // Real output is the first. Shuffle outputs
-    MTRACE(outs.back().size() << " outputs added. Real output is first. Shuffling. Global index order:");
-    std::shuffle(outs.back().begin(), outs.back().end(), std::default_random_engine(crypto::rand<unsigned>()));
+    MTRACE(outs.back().size() << " outputs added. Sorting outputs by index:");
+    std::sort(outs.back().begin(), outs.back().end(), [](const get_outs_entry &a, const get_outs_entry &b) { return std::get<0>(a) < std::get<0>(b); });
```

Comment:
Things like this should be squashed with the commit doing it wrong in the first place (assuming it's not a bug fix for existing code).